cmd/executor/cmd/root.go

```diff
@@ -110,21 +112,45 @@ func checkDockerfilePath() error {
 	return errors.New("please provide a valid path to a Dockerfile within the build context")
 }
 
-// resolveSourceContext unpacks the source context if it is a tar in a GCS bucket
+// resolveSourceContext unpacks the source context if it is a tar in a bucket
 // it resets srcContext to be the path to the unpacked build context within the image
 func resolveSourceContext() error {
 	if srcContext == "" && bucket == "" {
-		return errors.New("please specify a path to the build context with the --context flag or a GCS bucket with the --bucket flag")
+		return errors.New("please specify a path to the build context with the --context flag or a bucket with the --bucket flag")
 	}
-	if bucket == "" {
+
+	buildContextPath := constants.BuildContextDir
+	var contextExecutor buildcontext.BuildContext
+
+	if !strings.Contains(bucket, "://") {
 		return nil
 	}
-	logrus.Infof("Using GCS bucket %s as source context", bucket)
-	buildContextPath := constants.BuildContextDir
-	if err := util.UnpackTarFromGCSBucket(bucket, buildContextPath); err != nil {
+
+	// if no context is set, add default file context.tar.gz
+	if !strings.HasSuffix(bucket, ".tar.gz") {
+		bucket += "/" + constants.ContextTar
+	}
+
+	if strings.HasPrefix(bucket, "file://") {
```

Comment:
nit: could we make this dir:// since there may be multiple files in a directory being used as the build context

Reply:
When you look for a super fancy solution and it really is as simple as this.